pkg/webhooks/clusterrolebinding/clusterrolebinding.go

```diff
@@ -0,0 +1,200 @@
+}
+
+// renderSCC render the SCC object from the requests
+func (s *ClusterRoleBindingWebHook) renderClusterRoleBinding(request admissionctl.Request) (*rbacv1.ClusterRoleBinding, error) {
+	decoder, err := admissionctl.NewDecoder(&s.s)
+	if err != nil {
+		return nil, err
+	}
+	clusterRoleBinding := &rbacv1.ClusterRoleBinding{}
+
+	if len(request.OldObject.Raw) > 0 {
+		err = decoder.DecodeRaw(request.OldObject, clusterRoleBinding)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	return clusterRoleBinding, nil
+}
+
+// isProtectedNamespace returns true if clusterRoleBinding subject link
+// to ServiceAccount and openshift-*|kube-system ns
+func isProtectedNamespace(clusterRoleBinding *rbacv1.ClusterRoleBinding) bool {
+	isProtectNs := false
+
+	for _, subject := range clusterRoleBinding.Subjects {
+		if subject.Kind == "ServiceAccount" {
+			ns := subject.Namespace
+			if protectedNamespaces.Match([]byte(ns)) && !utils.SliceContains(ns, exceptionNamespaces) {
+				isProtectNs = true
+			}
+		}
+	}
+
+	return isProtectNs
+}
+
+// GetURI implements Webhook interface
+func (s *ClusterRoleBindingWebHook) GetURI() string {
+	return "/" + WebhookName
+}
+
+// Validate implements Webhook interface
+func (s *ClusterRoleBindingWebHook) Validate(request admissionctl.Request) bool {
+	valid := true
+	valid = valid && (request.UserInfo.Username != "")
+	valid = valid && (request.Kind.Kind == "ClusterRoleBinding")
+
+	return valid
+}
+
+// Name implements Webhook interface
+func (s *ClusterRoleBindingWebHook) Name() string {
+	return WebhookName
+}
+
+// FailurePolicy implements Webhook interface
+func (s *ClusterRoleBindingWebHook) FailurePolicy() admissionregv1.FailurePolicyType {
+	return admissionregv1.Ignore
+}
+
+// MatchPolicy implements Webhook interface
+func (s *ClusterRoleBindingWebHook) MatchPolicy() admissionregv1.MatchPolicyType {
+	return admissionregv1.Equivalent
+}
+
+// Rules implements Webhook interface
+func (s *ClusterRoleBindingWebHook) Rules() []admissionregv1.RuleWithOperations {
+	return rules
+}
+
+// ObjectSelector implements Webhook interface
+func (s *ClusterRoleBindingWebHook) ObjectSelector() *metav1.LabelSelector {
+	return nil
+}
+
+// SideEffects implements Webhook interface
+func (s *ClusterRoleBindingWebHook) SideEffects() admissionregv1.SideEffectClass {
+	return admissionregv1.SideEffectClassNone
+}
+
+// TimeoutSeconds implements Webhook interface
+func (s *ClusterRoleBindingWebHook) TimeoutSeconds() int32 {
+	return timeout
+}
+
+// Doc implements Webhook interface
+func (s *ClusterRoleBindingWebHook) Doc() string {
+	return fmt.Sprintf(docString, managedNamespaces)
+}
+
+// SyncSetLabelSelector returns the label selector to use in the SyncSet.
+// Return utils.DefaultLabelSelector() to stick with the default
+func (s *ClusterRoleBindingWebHook) SyncSetLabelSelector() metav1.LabelSelector {
+	return utils.DefaultLabelSelector()
+}
+
+func (s *ClusterRoleBindingWebHook) HypershiftEnabled() bool { return true }
```

Comment:
I think you will need to run `make package` to generate package manifests for hypershift

Reply:
Thanks @bmeng , generated the mainfests for HS

Reply:
@mrbarge @ravitri  Any of you know if we have criteria for adding webhook to hypershift clusters?

Reply:
Nothing documented. I think it's a case-by-case basis of asking the question:
- could a cluster owner even do this action from their hosted cluster? (if no: then not needed)
- if a cluster owner does this action from their hosted cluster, would it potentially bring the cluster into an unsupportable state? (if yes, probably needed)

In this case I think if we want this protection for OSD/ROSA it makes sense to also protect it in hosted clusters.